### PR TITLE
Add html char code tool

### PR DESCRIPTION
**Pull request by shu1007**
## Type of Pull Request

- [ ] Bug fix
- [x] Enhancement

Related Issue #s or links (if any):  #48

## Description of Changes

I added HTML Character Codes page. It  encode and decode HTML special characters .

### CODE REVIEW

File: pages/html-char-codes.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import Heading from '../components/Heading';
+import HtmlCharCodes from '../components/html-char-code/HtmlCharCode';
+import Layout from '../components/Layout';
+
+export default function HtmlCharCodesPage() {
+  return (
+    <Layout title='HTML Character Codes'>
+      <Heading>HTML Character Codes</Heading>
+      <Typography paragraph>
+        Paste or type in some text and it will be encode or decode all
```
Comment (dlford) [inline pages/html-char-codes.tsx:13]: Minor issue, "Paste or type in some text and it will encode or decode all HTML character codes"

File: components/TextFieldWithCopyOrPaste.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    );
+  }
+
+  return (
+    <Box
+      display='flex'
+      flexDirection='column'
+      gap={2}
+    >
+      <TextField
```
Comment (dlford) [inline components/TextFieldWithCopyOrPaste.tsx:77]: Lots of clever techniques in here, and kudos for that, unfortunately it does not meet the requirements of the original issue.

**Review by dlford — CHANGES_REQUESTED** [review]
@shu1007 Great job on this! But it does not currently meet the requirements laid out in the original issue.

This PR should... `Provide an "unescaped" and "escaped" field that auto-fills like the Base64 tool and is responsive for mobile devices`

Essentially, I don't think the mode selector adds anything useful here, I think it's a much better UX to have two clearly labelled fields that you can copy/paste/clear or type into to run the conversion without parsing and manipulating a select element.

**Comment by shu1007** [comment]
@dlford 
Sorry. I fixed.
Please review again.

File: data/nav.ts [hunk 3]
```diff
@@ -74,6 +75,11 @@ const navItems = [
     href: '/regex',
     Icon: Search,
   },
+  {
+    title: 'Html Character Codes',
```
Comment (dlford) [inline data/nav.ts:79]: Can we shorten this to `HTML Char Codes`?

File: components/html-char-code/HtmlCharCode.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        hasCopy
+        label='Escaped'
+        value={escaped}
+        onClearClick={clearBoth}
+        onPasteCleck={(text) => {
+          setEscaped(text);
+          setUnescaped(decode(text));
+        }}
+        onChange={(event) => setEscaped(event.target.value)}
+        onBlur={(e) => {
```
Comment (dlford) [inline components/html-char-code/HtmlCharCode.tsx:56]: Let's move this to `onChange` so both inputs work the same. If needed we can add user facing error message when the input is causing an error.

File: components/TextFieldWithCopyPaste.tsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+import ContentPasteGoIcon from '@mui/icons-material/ContentPasteGo';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import React, { useState } from 'react';
+
+import useSupportsClipboardRead from '../hooks/useSupportsClipboardRead';
+import Toast, { ToastProps } from './Toast';
+
+type Props = {
```
Comment (dlford) [inline components/TextFieldWithCopyPaste.tsx:12]: Since this is a generic component, lets make this an interface and export it for later use elsewhere.

**Review by dlford — CHANGES_REQUESTED** [review]
@shu1007 no worries! Much better, just a few minor details. thanks!

File: components/html-char-code/HtmlCharCode.tsx (hunk 4 above)
Comment (shu1007) [reply components/html-char-code/HtmlCharCode.tsx]: I was mistakenly thinking that onChange would cause the other onChange to fire.
But there was no problem, so we fixed it!
